### PR TITLE
Trigger E_USER_DEPRECATED for deprecated features

### PR DESCRIPTION
**Pull request by malarzm**
I think I've covered all places that make sense. Not sure about all types annotations - should each deprecated one trigger error in its constructor?

### CODE REVIEW

File: lib/Doctrine/ODM/MongoDB/EagerCursor.php [hunk 1]
```diff
@@ -18,6 +18,8 @@
  */
 
 namespace Doctrine\ODM\MongoDB;
+use Doctrine\MongoDB\CursorInterface;
```
Comment (alcaeus) [inline lib/Doctrine/ODM/MongoDB/EagerCursor.php]: I believe we've previously used a blank line before `use`.

Comment (malarzm) [inline lib/Doctrine/ODM/MongoDB/EagerCursor.php]: Yes we did, must have overlooked Storm adding `use` in the first place :+1:

**Comment by malarzm** [comment]
> should each deprecated one trigger error in its constructor?

Unfortunately it's not possible as [`Annotation::__construct` is final`](https://github.com/doctrine/annotations/blob/7c830a576f9471f71096d200d633b6400513ccfb/lib/Doctrine/Common/Annotations/Annotation.php#L44) :( Next idea is hooking into `AnnotationDriver` and detect when there's deprecated annotation used, thoughts @alcaeus?

**Comment by alcaeus** [comment]
Can we do away with [DoctrineAnnotations.php](https://github.com/doctrine/mongodb-odm/blob/master/lib/Doctrine/ODM/MongoDB/Mapping/Annotations/DoctrineAnnotations.php) and have the annotations autoloaded? In that case you could put the deprecation notice in the file itself. If not, we'll have to handle it in the driver. In that case, we could add a deprecation message to the annotation class itself by using a `DEPRECATED` constant (along with the usual `@deprecated` doc) and check against that in the driver. Hacky, but it avoids keeping a map of deprecated annotations and their messages in the driver.

**Comment by malarzm** [comment]
Not really as that would be a BC break, there's [AnnotationDriver::registerAnnotationClasses](https://github.com/doctrine/mongodb-odm/blob/80e8df78856372136734ec6920582d855703796e/lib/Doctrine/ODM/MongoDB/Mapping/Driver/AnnotationDriver.php#L52) which was recommended to use. I guess driver it is

**Comment by alcaeus** [comment]
> I guess driver it is

I guess driver it is then.

**Comment by malarzm** [comment]
@alcaeus annotations done, also added 2 more notices I forgot about before.

**Comment by malarzm** [comment]
> Note: apparently the email address associated with your first commit isn't associated with your GitHub account, you might want to take a look at your configs ;)

Woah nice catch! Changed that one to be public me and claimed that email address, must have came from my workstation :)

**Comment by alcaeus** [comment]
> Woah nice catch!

Saw your picture twice in the commit list. Once is plenty enough 😜

**Pull request merged**